src/tdi_init.cpp

```diff
@@ -110,6 +121,7 @@ tdi_status_t DevMgr::deviceIdListGet(
 }
 
 tdi_status_t DevMgr::deviceRemove(const tdi_dev_id_t &dev_id) {
+  removeItemFromMap(&this->dev_map_, dev_id);
```

Comment:
do we need such a complex function and loop over every entry in the map like this? I think just a simple `this->dev_map_.erase(dev_id)` should be enough. Probably do a find first, and then use the iterator if not nullptr to do the erase so that no exception is thrown